inference-engine/tests/functional/plugin/conformance/subgraphs_dumper/src/matchers/single_op.cpp

```diff
@@ -0,0 +1,184 @@
+// Copyright (C) 2018-2019 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "matchers/single_op.hpp"
+#include "ngraph/ops.hpp"
+
+using namespace SubgraphsDumper;
+
+// TODO: Move to some utils?
+bool compare_constants_data(const std::shared_ptr<ngraph::op::Constant> &op,
+                            const std::shared_ptr<ngraph::op::Constant> &ref) {
+    switch (op->get_element_type()) {
+        case ngraph::element::Type_t::boolean:
+            if (op->cast_vector<bool>() != ref->cast_vector<bool>()) {
+                return false;
+            } else {
+                return true;
+            }
```

Comment:
IMHO `if` is redundant. Why not:
```
return op->cast_vector<bool>() == ref->cast_vector<bool>()
```

Reply:
Hmm, don't know why I wrote it so, agree with you